encoding/ccf/ccf_type_id.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2022-2023 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ccf
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/onflow/cadence"
+)
+
+// ccfTypeID represents CCF type ID.
+type ccfTypeID uint64
+
+func newCCFTypeID(b []byte) ccfTypeID {
+	return ccfTypeID(new(big.Int).SetBytes(b).Uint64())
```

Comment:
What if we use `binary.BigEndian.Uint64()` here? would that save extra allocations of  `new(big.Int)`.
Same question on the `Bytes()` method.

Reply:
Bastian asked a similar question. :smile:

There are two reasons for using `big.Int` under the hood even though `ccfTypeID` is `uint64`:
- `ccfTypeID` can be changed to `big.Int` type if we decide to use hash digest as CCF ID (less important).
- using `big.Int` allows smaller encoded size vs `binary.BigEndian` which is always 8 bytes (more important).

Also `new(big.Int)` here and in `Bytes()` don't escape to heap.

I tried a custom function without `big.Int` to encode to same small size, but there was no difference in memory and allocations.  Speed was also comparable as well.